Userland/Applications/PixelPaint/BucketTool.cpp

```diff
@@ -47,8 +48,12 @@ static void flood_fill(Gfx::Bitmap& bitmap, Gfx::IntPoint const& start_position,
 
     Queue<Gfx::IntPoint> queue;
     queue.enqueue(start_position);
+    HashTable<Gfx::IntPoint> visited;
     while (!queue.is_empty()) {
         auto position = queue.dequeue();
+        if (visited.contains(position))
+            continue;
+        visited.set(position);
```

Comment:
Wouldn't it be more {time,memory}-efficient to not add duplicates to the queue in the first place, instead of only ignoring them when going though all entries?

Reply:
I have not run any benchmarks, but I suspect you would be right. This was just the cleaner code approach so I went with this. I can investigate if there's any sort of noticable difference.

Reply:
Upon some messing around, I can't seem to find any significant difference in either way of doing it. (Though perhaps this is just my computer being wonky, serenity is very laggy to begin with, I don't get more than 10fps I would say). 

I can make the change if you think it's more appropriate, but the change isn't as nice as the current one:
```cpp
  if (position.x() != 0) {
      auto neighbor = position.translated(-1, 0);
      if (!visited.contains(neighbor))
          queue.enqueue(neighbor);
  }

  if (position.x() != bitmap.width() - 1) {
      auto neighbor = position.translated(1, 0);
      if (!visited.contains(neighbor))
          queue.enqueue(neighbor);
  }

  if (position.y() != 0) {
      auto neighbor = position.translated(0, -1);
      if (!visited.contains(neighbor))
          queue.enqueue(neighbor);
  }

  if (position.y() != bitmap.height() - 1) {
      auto neighbor = position.translated(0, 1);
      if (!visited.contains(neighbor))
          queue.enqueue(neighbor);
  }
```